### ISSUE TITLE
It generate all the same sentences to different images, and i found the input of LSTM in the second time step is all zeros.

### ISSUE THREAD

**Issue by LaiPiXiong** [ruotianluo/ImageCaptioning.pytorch#50]
It generate all the same sentences when eval. I found when sample_max == 1, the second input to LSTM is all zeros. And if sample_max != 1, the result is not match to the picture although it seems right.

**Comment by ruotianluo** [comment]
can you be more specific? What model are you using? which variable is zero?

**Comment by LaiPiXiong** [comment]
@ruotianluo Hi, The model is ShowTellModel（without attention）. In LSTM, the feature map is first input to(I checked all the features is OK), but the output of LSTM's first time step, becomes all the same(before sotfmax). I try a lot, and i think self.core is not work. It seems like no matter what i feed in, the output is fixed.

**Comment by LaiPiXiong** [comment]
There is a “sample_max True” branch which confused me much for it is not mutex to “ t==1” branch. And another question, in  test case,  do we need to feed in zeros when “t ==1”as “bos” tag? Because in my opinion in test phrase, the model will generate words automatically,  we just need to judge when to stop. So many

**Comment by ruotianluo** [comment]
We do need to manually feed zero. think of that as a signal to tell the model to generate sentence.

**Comment by LaiPiXiong** [comment]
@ruotianluo Hi, I use inception model to generate features instead of the pretrained fc_features. And till now it still generate a same caption for all images in a  batch. (Now the train loss is about 2.56, test loss is 2.73). What can I do for this situation? 
PS: the caption is always the same from start to now.

**Comment by ruotianluo** [comment]
The loss looks correct.  There should be some issue during sampling. Try debug step by step.
